alg/teca_deeplabv3p_ar_detect.py

```diff
@@ -1,309 +1,314 @@
-class Bottleneck(nn.Module):
-    expansion = 4
-
-    def __init__(self, inplanes, planes, stride=1, rate=1, downsample=None):
-        super(Bottleneck, self).__init__()
-        self.conv1 = nn.Conv2d(inplanes, planes, kernel_size=1, bias=False)
-        self.bn1 = nn.BatchNorm2d(planes)
-        self.conv2 = nn.Conv2d(planes, planes, kernel_size=3, stride=stride,
-                               dilation=rate, padding=rate, bias=False)
-        self.bn2 = nn.BatchNorm2d(planes)
-        self.conv3 = nn.Conv2d(planes, planes * 4, kernel_size=1, bias=False)
-        self.bn3 = nn.BatchNorm2d(planes * 4)
-        self.relu = nn.ReLU(inplace=True)
-        self.downsample = downsample
-        self.stride = stride
-        self.rate = rate
-
-    def forward(self, x):
-        residual = x
-
-        out = self.conv1(x)
-        out = self.bn1(out)
-        out = self.relu(out)
-
-        out = self.conv2(out)
-        out = self.bn2(out)
-        out = self.relu(out)
-
-        out = self.conv3(out)
-        out = self.bn3(out)
-
-        if self.downsample is not None:
-            residual = self.downsample(x)
-
-        out += residual
-        out = self.relu(out)
-
-        return out
-
-
-class ResNet(nn.Module):
-    def __init__(self, nInputChannels, block, layers, os=16):
-        self.inplanes = 64
-        super(ResNet, self).__init__()
-        if os == 16:
-            strides = [1, 2, 2, 1]
-            rates = [1, 1, 1, 2]
-            blocks = [1, 2, 4]
-        elif os == 8:
-            strides = [1, 2, 1, 1]
-            rates = [1, 1, 2, 2]
-            blocks = [1, 2, 1]
-        else:
-            raise NotImplementedError
-
-        # Modules
-        self.conv1 = nn.Conv2d(nInputChannels, 64, kernel_size=7,
-                               stride=2, padding=3, bias=False)
-        self.bn1 = nn.BatchNorm2d(64)
-        self.relu = nn.ReLU(inplace=True)
-        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
-
-        self.layer1 = self._make_layer(block, 64, layers[0],
-                                       stride=strides[0], rate=rates[0])
-        self.layer2 = self._make_layer(block, 128, layers[1],
-                                       stride=strides[1], rate=rates[1])
-        self.layer3 = self._make_layer(block, 256, layers[2],
-                                       stride=strides[2], rate=rates[2])
-        self.layer4 = self._make_MG_unit(block, 512, blocks=blocks,
-                                         stride=strides[3], rate=rates[3])
-
-        self._init_weight()
-
-    def _make_layer(self, block, planes, blocks, stride=1, rate=1):
-        downsample = None
-        if stride != 1 or self.inplanes != planes * block.expansion:
-            downsample = nn.Sequential(
-                nn.Conv2d(self.inplanes, planes * block.expansion,
-                          kernel_size=1, stride=stride, bias=False),
-                nn.BatchNorm2d(planes * block.expansion),
+from teca_py import *
```

Comment:
this seems to be a left over from your experiments with the recursive import. please back this one out too.